treeherder/perf/perf_sheriff_bot.py

```diff
@@ -0,0 +1,20 @@
+from datetime import datetime
+from typing import List
+
+from treeherder.perf.alerts import (AlertsPicker,
+                                    BackfillReport,
+                                    BackfillReportMaintainer,
+                                    IdentifyAlertRetriggerables)
+
+
+class PerfSfheriffBot:
+    """
+    Wrapper class used to aggregate the reporting of backfill reports.
+    """
+    def __init__(self, alerts_picker: AlertsPicker, backfill_context_fetcher: IdentifyAlertRetriggerables):
+        self.reportMaintainer = BackfillReportMaintainer(alerts_picker, backfill_context_fetcher)
+
+    def report(self, since: datetime,
```

Comment:
We could rename `report()` to something more generic, to hint the future interface that'll do all the work (reporting, backfilling & secretary-ing).

Reply:
I think it would be the best to keep `report` as it is now, and add a new `backfill` method for backfill. If these method need to be unified in the future, they can be encapsulated into a new method.
I bet this would be clearer as we implement the bot.